### PR TITLE
Use nmcli instead of ifupdown

### LINKED ISSUE #321: Switch to Network Manager (nmcli) for network configuration on Ubuntu
#319 covered a problem when DNS settings were not taking effect via `/etc/network/interfaces` (though the custom Docksal subnet was created as expected).

Using `/etc/network/interfaces` for network configuration on Ubuntu is deprecated.

Network Manager is the current way of managing network connections with a cli tool called `nmcli`
https://www.tecmint.com/configure-network-connections-using-nmcli-tool-in-linux/

Looks like `nmcli` is a standard tool across Debian family distros, so we should have the unofficial compatibility across the Debian family preserved.

### PR DESCRIPTION
**Pull request by ohthehugemanatee**
closes #321 

Switch to nmcli instead of modifying /etc/network/interfaces. This is my first time using nmcli, so please someone check my work!

### CODE REVIEW

**Comment by achekulaev** [comment]
Related #384

**Comment by lmakarov** [comment]
On a clean Ubuntu 17.10 (DigitalOcean) neither `ifup`/`ifdown`, nor `nmcli` are available out of the box:

```
root@ubuntu-512mb-sfo2-01:~# ifup
The program 'ifup' can be found in the following packages:
 * ifupdown
 * ifupdown2
 * netscript-2.4
Try: apt install <selected package>
root@ubuntu-512mb-sfo2-01:~# nmcli
The program 'nmcli' is currently not installed. You can install it by typing:
apt install network-manager
```

We will need to check if `nmcli` is available and do `sudo apt install network-manager` otherwise.

**Comment by lmakarov** [comment]
After installing network-manager it has to be started before `nmcli` can be used:

```
root@ubuntu-512mb-sfo2-01:~# nmcli
Error: NetworkManager is not running.
root@ubuntu-512mb-sfo2-01:~# /etc/init.d/network-manager start
[ ok ] Starting network-manager (via systemctl): network-manager.service.
root@ubuntu-512mb-sfo2-01:~# nmcli dev status
DEVICE  TYPE      STATE      CONNECTION 
eth0    ethernet  unmanaged  --         
lo      loopback  unmanaged  --  
```

File: bin/fin [hunk 1]
```diff
@@ -2650,15 +2650,10 @@ configure_resolver_linux () {
...
+  if ! grep -q 'docksal' `nmcli dev status`; then
 		echo-green "Adding a subnet for Docksal..."
-		cat > /tmp/docksal.ip.addr <<EOF
-	up   ip addr add ${DOCKSAL_DEFAULT_SUBNET} dev lo label lo:docksal
-	down ip addr del ${DOCKSAL_DEFAULT_SUBNET} dev lo label lo:docksal
-	dns-nameservers ${DOCKSAL_DEFAULT_IP}
-EOF
-		sudo sed -i '/iface lo inet loopback/r /tmp/docksal.ip.addr' /etc/network/interfaces
-		rm -f /tmp/docksal.ip.addr
+    nmcli con add con-name lo:docksal ifname docksal0 type bridge ip4 ${DOCKSAL_DEFAULT_SUBNET}
```
Comment (lmakarov) [inline bin/fin]: I don't think this works as anticipated:

```
root@ubuntu-512mb-sfo2-01:~# nmcli con add con-name lo:docksal ifname docksal0 type bridge ip4 192.168.64.1/24
Connection 'lo:docksal' (79ff1fa9-674d-4ee8-8072-185022c9cc0d) successfully added.
root@ubuntu-512mb-sfo2-01:~# nmcli con modify lo:docksal ipv4.dns 192.168.64.100
```

I see the settings are added in `/etc/network/interfaces`:

```
root@ubuntu-512mb-sfo2-01:~# cat /etc/network/interfaces 
# This file describes the network interfaces available on your system
# and how to activate them. For more information, see interfaces(5).

# The loopback network interface
auto lo
iface lo inet loopback
	up   ip addr add 192.168.64.1/24 dev lo label lo:docksal
	down ip addr del 192.168.64.1/24 dev lo label lo:docksal
	dns-nameservers 192.168.64.100

source /etc/network/interfaces.d/*
```

Unfortunately these do not get applied right away, neither after a reboot, so neither of these IPs can be reached:

```
192.168.64.1
192.168.64.100
```

File: bin/fin [hunk 2]
```diff
@@ -2650,15 +2650,10 @@ configure_resolver_linux () {
...
+    nmcli con modify lo:docksal ipv4.dns ${DOCKSAL_DEFAULT_IP}
 		sudo ifdown --force lo 2>/dev/null
```
Comment (lmakarov) [inline bin/fin]: `ifup`, `ifdown` and `resolvconf` are not available out of the box. Should be either installed or replaced with an alternative command available out out the box (preferable).

File: bin/fin (hunk 1 above)
Comment (lmakarov) [reply bin/fin]: Scratch the part about settings in `/etc/network/interfaces`. Those were added after I ran Docksal installation...

File: bin/fin (hunk 2 above)
Comment (achekulaev) [reply bin/fin]: `ifup`/`ifdown` is a very default utility. Here is empty Ubuntu Xenial from Scaleway.

```
root@scw-eafd0d:~# which ifup
/sbin/ifup
```

Comment (achekulaev) [reply bin/fin]: Ok nevermind. After Xenial ifup is no longer available by default... Thanks Ubuntu!

**Comment by lmakarov** [comment]
Closing this, as it looks like all we need on recent Ubuntu versions ([confirmed](https://github.com/docksal/docksal/issues/384#issuecomment-349012108) with 17.10) is:

```
sudo apt-get install ifupdown resolvconf -y
```

**Pull request closed without merging**